### DISCUSSION TITLE
Doing a plot only with currents (with no seeded elements)

### DISCUSSION THREAD

**Discussion by RodrigoCC11** [OpenDrift/opendrift#1374] [Q&A] [answered]
Hi! I'm trying to do a plot using o.plot() function and with x and y surface current velocity on the background, but I don't want to seed any element, I really just want the grid of the current. I tried to do some research but I haven't found anything yet. I was hoping someone could help me figure out what I should do. I also would like to do an animation of the currents over time. Is there any function ou functions that help me do it? Thanks!

**Comment by knutfrode** [comment, marked as answer]
Hi,

You can plot a scalar field from a reader like this:
`r.plot(variable='sea_water_temperature', time=datetime(2024, 8, 1, 12))`
However, this does not yet support vector fields (e.g. current), and there is no reader method to create animations.

But you can make both plots and animations from a simulation object, after seeding an element that you can chose not to show:
```
from datetime import datetime, timedelta
from opendrift.models.oceandrift import OceanDrift

o = OceanDrift()
o.add_readers_from_list(['https://thredds.met.no/thredds/dodsC/sea/norkyst800m/1h/aggregate_be'])
o.seed_elements(lon=4, lat=60, time=datetime(2024, 8, 10))
o.run(time_step_output=timedelta(hours=1), end_time=datetime(2024, 8, 11))
o.animation(background=['x_sea_water_velocity', 'y_sea_water_velocity'], clabel='Surface current [m/s]',
            show_elements=False,
            corners=[4.5, 5.5, 60, 60.5],
            filename='current.mp4')
```

The `plot` and `animation` methods have a lot of parameters that you can play with, e.g. `skip` and `scale` to change the density of displayed vectors.
https://opendrift.github.io/autoapi/opendrift/models/basemodel/index.html#opendrift.models.basemodel.OpenDriftSimulation.plot
https://opendrift.github.io/autoapi/opendrift/models/basemodel/index.html#opendrift.models.basemodel.OpenDriftSimulation.animation

![current](https://github.com/user-attachments/assets/f3603699-09ba-4588-a4cd-0ef14a88520b)

**Reply by RodrigoCC11** [reply]
Thank you very much for your help! I understood it all! Will certainly explore the documentation.
